# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: testing/marionette/print.js
# Commit: 909001ac06d8
# Full Hash: 909001ac06d8b3982116a0b0f753ae365b1cea22
# Author: Jonathan Watt <jwatt@jwatt.org>
# Date: 2020-07-23 03:20:25
# Regressor Bug: 1652270
# File Overlap Count: 1
# Description:
#   Bug 1652270. Convert nsFrameLoader::Print to return a Promise. r=farre,remote-protocol-reviewers,marionette-reviewers,jgraham,whimboo,mixedpuppy
#   
#   (Instead of requiring callers to pass an nsIWebProgressListener.)
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D83249
# ==============================================================================

diff -r 8cfbd5213d04 -r 909001ac06d8 testing/marionette/print.js
--- a/testing/marionette/print.js	Thu Jul 23 02:01:54 2020 +0300
+++ b/testing/marionette/print.js	Wed Jul 22 23:17:45 2020 +0000
@@ -104,36 +104,22 @@
 
   let printSettings = getPrintSettings(settings, filePath);
 
+  await browser.print(outerWindowID, printSettings);
+
+  // Bug 1603739 - With e10s enabled the promise returned by print() resolves
+  // too early, which means the file hasn't been completely written.
   await new Promise(resolve => {
-    // Bug 1603739 - With e10s enabled the WebProgressListener states
-    // STOP too early, which means the file hasn't been completely written.
-    const waitForFileWritten = () => {
-      const DELAY_CHECK_FILE_COMPLETELY_WRITTEN = 100;
+    const DELAY_CHECK_FILE_COMPLETELY_WRITTEN = 100;
 
-      let lastSize = 0;
-      const timerId = setInterval(async () => {
-        const fileInfo = await OS.File.stat(filePath);
-        if (lastSize > 0 && fileInfo.size == lastSize) {
-          clearInterval(timerId);
-          resolve();
-        }
-        lastSize = fileInfo.size;
-      }, DELAY_CHECK_FILE_COMPLETELY_WRITTEN);
-    };
-
-    const printProgressListener = {
-      onStateChange(webProgress, request, flags, status) {
-        if (
-          flags & Ci.nsIWebProgressListener.STATE_STOP &&
-          flags & Ci.nsIWebProgressListener.STATE_IS_NETWORK
-        ) {
-          waitForFileWritten();
-        }
-      },
-      QueryInterface: ChromeUtils.generateQI(["nsIWebProgressListener"]),
-    };
-
-    browser.print(outerWindowID, printSettings, printProgressListener);
+    let lastSize = 0;
+    const timerId = setInterval(async () => {
+      const fileInfo = await OS.File.stat(filePath);
+      if (lastSize > 0 && fileInfo.size == lastSize) {
+        clearInterval(timerId);
+        resolve();
+      }
+      lastSize = fileInfo.size;
+    }, DELAY_CHECK_FILE_COMPLETELY_WRITTEN);
   });
 
   logger.debug(`PDF output written to ${filePath}`);